### DISCUSSION TITLE
Getting the currently authenticated user's profile?

### DISCUSSION THREAD

**Discussion by jamalmoir** [tweepy/tweepy#1711] [Questions] [answered]
Is there a good way to do this?

I have seen mentions of `API.me()` searching around, but it doesn't seem to exist in the source code anymore.

The OAuthHandler's `username` field is also defaulted to `None` and never updated.

The best way I can think of is grabbing the user's timeline, finding the user's id or username in that, then using `API.get_user()`.

Surely there is a better way than this?

**Comment by Harmon758** [comment, marked as answer]
You can use [`API.verify_credentials`](https://tweepy.readthedocs.io./en/v4.4.0/api.html#tweepy.API.verify_credentials).
